Relnotes for June release (#932)
MicrosoftDocs/quantum-docs-pr: articles/resources/relnotes.md

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

> ## Version 0.xx.xxxx.xxx

Reviewer: The version number will be in a slightly different format this time to bring it in line with modern NuGet versioning practices (namely, using SemVer 2.0). In particular, the version number will look like `0.12.xxxxxxxx` (no third separator).

Author: sounds good, let's update this when we have a version number

*Release date: July 1st, 2020*

This release contains the following:

> - New qdk-chem chemistry format conversion tool 

Reviewer: I forget the exact uid, but perhaps a link like the following would be helpful? Perhaps it would be good to have a brief page on how to install and use `qdk-chem` as well? Thoughts, @KittyYeungQ?

```suggestion
- New `qdk-chem` tool for converting legacy electronic structure problem serialization formats (e.g.: FCIDUMP) to [Broombridge](xref:todo)
```

> - New synthesis functionality in QuantumLibraries

Reviewer: ```suggestion
- New functions and operations in the [`Microsoft.Quantum.Synthesis` namespace](xref:microsoft.quantum.synthesis) for coherently applying classical oracles using transformation- and decomposition-based synthesis algorithms.
```

> - IQ# now allows arguments to %simulate and %estimate

Reviewer: ```suggestion
- IQ# now allows arguments to the `%simulate`, `%estimate`, and other magic commands. See the [`%simulate` magic command reference](microsoft.quantum.iqsharp.magic-ref.simulate) for more details.
```

> - New phase display options in IQ#

Reviewer: ```suggestion
- New phase display options in IQ#. See the [`%config` magic command reference](xref:microsoft.quantum.iqsharp.magic-ref.config) for more details.
```

> - Updated conda package support for IQ# installation to support local installation to a conda environment

Reviewer: @rmshaffer, @KittyYeungQ: Should we have a brief documentation page on this feature, perhaps marked as an advanced page or something?

```suggestion
- IQ# is now provided as a [`conda` package](https://anaconda.org/quantum-engineering/iqsharp) for IQ# installation to support local installation to a conda environment.
```

Reviewer 2: I've actually been working on updating the IQ#/Python installation guide to make conda the recommended path, and mark the existing one as "advanced". Here's preview renderings of what I have so far:

- https://review.docs.microsoft.com/en-us/quantum/quickstarts/install-jupyter?branch=pr-en-us-923&tabs=tabid-conda
- https://review.docs.microsoft.com/en-us/quantum/quickstarts/install-python?branch=pr-en-us-923&tabs=tabid-conda
- https://review.docs.microsoft.com/en-us/quantum/quickstarts/update?branch=pr-en-us-923&tabs=tabid-conda

Draft PR here: https://github.com/MicrosoftDocs/quantum-docs-pr/pull/923

> - When using the simulator, qubits no longer need to be in a 0 state upon release and instead can be measured or in a 0 state

Reviewer: I think this is broader than just the simulator, actually. @swernli and @bettinaheim may be able to confirm and suggest improved wording here.
```suggestion
- When using the simulator, qubits no longer need to be in the |0⟩ state upon release, but can be automatically reset if they were measured immediately before releasing.
```

> - Bug fixes 

Reviewer: We should mention the new version roll-forward behavior here, especially since that has such an impact on the katas.

Author: any suggested language? What's the user impact to highlight here?

Reviewer: The old behavior was that in order to use two Q# libraries together, they had to be built with the exact same version of the Quantum Development Kit. For the most part, that wasn't a huge problem given that the Quantum Development Kit isn't installed globally, but is referenced as an SDK from each csproj. When using IQ#, however, that meant that you needed the globally installed version of IQ# to match the package you wanted to use it with, something that was especially annoying for katas users running locally rather than online or in containers.

This change relaxes that so only the major and minor version have to match, making it much easier for IQ# users to make use of Q# library packages like the katas.

@rmshaffer's changes to allow implicit version numbers for the katas package and to allow conda environment–local rather than machine-local installations of IQ# also help this considerably, such that this release makes it much easier for users to consume libraries from IQ#.

+@tcNickolas for katas impact.

See the full list of closed PRs for [libraries](https://github.com/Microsoft/QuantumLibraries/pulls?q=is%3Apr+is%3Aclosed), [compiler](https://github.com/microsoft/qsharp-compiler/pulls?q=is%3Apr+is%3Aclosed), [runtime](https://github.com/microsoft/qsharp-runtime/pulls?q=is%3Apr+is%3Aclosed), [samples](https://github.com/Microsoft/Quantum/pulls?q=is%3Apr+is%3Aclosed), [IQ#](https://github.com/microsoft/iqsharp/pulls?q=is%3Apr+is%3Aclosed) and [Katas](https://github.com/microsoft/QuantumKatas/pulls?q=is%3Apr+is%3Aclosed).  

> ## Version 0.11.2006.403

Reviewer: ```suggestion

## Version 0.11.2006.403
```

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

## Version 0.12.xxxxxxx

*Release date: July 1st, 2020*

This release contains the following:

- New `qdk-chem` tool for converting legacy electronic structure problem serialization formats (e.g.: FCIDUMP) to [Broombridge](xref:todo)

> - New functions and operations in the [`Microsoft.Quantum.Synthesis` namespace](xref:microsoft.quantum.synthesis) for coherently applying classical oracles using transformation- and decomposition-based synthesis algorithms.

Reviewer 3: The link to `microsoft.quantum.synthesis` is broken. This may be since the auto generated files have not been merged yet. The M.Q.Synthesis namespace does not show up in navigation of the preview of this PR.

Reviewer: That link will be broken until the docs are uploaded with this release, but you can check that the UID is correct from the beta builds of the docs (e.g.: https://github.com/MicrosoftDocs/quantum-docs-pr/blob/beta/v0.12.2006.2435/api/qsharp/Microsoft.Quantum.Synthesis.yml from two days ago).

> - IQ# now allows arguments to the `%simulate`, `%estimate`, and other magic commands. See the [`%simulate` magic command reference](microsoft.quantum.iqsharp.magic-ref.simulate) for more details.

Reviewer 2: ```suggestion
- IQ# now allows arguments to the `%simulate`, `%estimate`, and other magic commands. See the [`%simulate` magic command reference](xref:microsoft.quantum.iqsharp.magic-ref.simulate) for more details.
```

- New phase display options in IQ#. See the [`%config` magic command reference](xref:microsoft.quantum.iqsharp.magic-ref.config) for more details.

> - IQ# is now provided as a [`conda` package](https://anaconda.org/quantum-engineering/iqsharp) for IQ# installation to support local installation to a conda environment.

Reviewer 2: ```suggestion
- IQ# and the `qsharp` Python package are now provided via conda packages ([qsharp](https://anaconda.org/quantum-engineering/qsharp) and [iqsharp](https://anaconda.org/quantum-engineering/iqsharp)) to simplify local installation of Q# Jupyter and Python functionality to a conda environment. See the [Q# Jupyter Notebooks](https://docs.microsoft.com/quantum/quickstarts/install-jupyter) and [Q# with Python](https://docs.microsoft.com/quantum/quickstarts/install-python) installation guides for more details.
```
